testdata/ingest_memtable_overlaps

```diff
@@ -118,3 +118,21 @@ e-d
 true
 false
 true
+
```

Comment:
should we include these cases in [ingest_test.go](https://github.com/cockroachdb/pebble/blob/master/ingest_test.go#L495) as well? since that test works with an actual DB and we get to do a `DB.CheckLevels`